### PR TITLE
Add SHA384 and SHA512 to the STANDARD flavor of the crypto binary

### PR DESCRIPTION
**Pull request by kenlautner**
## Description

Authenticated variables were updated to allow the choice of SHA256, SHA384 or SHA512 for hashing.  To accommodate this change, we need the algorithms available for use.  This change updates the STANDARD flavor of shared crypto to include the SHA384 and SHA512 algorithms so that they can be used by consumers of the binary.

There is no size impact for the STANDARD flavor because all of the SHA implementations are built together in openssl.  This simply enables their use.

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested on locally generated binaries. SHA384 and SHA512 were available to use.

## Integration Instructions

N/A

### CODE REVIEW

**Review by kuqin12 — APPROVED** [review]

**Comment by kuqin12** [comment]
Will this get into the 2302 branches?

**Review by makubacki — APPROVED** [review]

File: CryptoPkg/Driver/Packaging/generate_cryptodriver.py
Comment (makubacki) [inline CryptoPkg/Driver/Packaging/generate_cryptodriver.py:1]: Can you please provide the `STANDARD` flavor size delta in the PR description?

File: CryptoPkg/Driver/Packaging/generate_cryptodriver.py [hunk 1]
```diff
@@ -143,7 +143,7 @@ def get_flavors():
             "guid": "d9a75606-caba-4aa0-80a6-591852335400"
```
Comment (kuqin12) [inline CryptoPkg/Driver/Packaging/generate_cryptodriver.py:143]: Do you need to touch "TINY_SHA" to include "SHA512"?

Comment (kuqin12) [reply CryptoPkg/Driver/Packaging/generate_cryptodriver.py:143]: Also, do you know if we still have any use cases for "SM3"?

**Review by kuqin12 — APPROVED** [review]

File: CryptoPkg/Driver/Packaging/generate_cryptodriver.py (hunk 1 above)
Comment (kenlautner) [reply CryptoPkg/Driver/Packaging/generate_cryptodriver.py:143]: We don't have any use cases for SM3 for any current platform.

File: CryptoPkg/Driver/Packaging/generate_cryptodriver.py
Comment (kenlautner) [reply CryptoPkg/Driver/Packaging/generate_cryptodriver.py:1]: Updated.

**Pull request merged**